Uncaught Error Error: Cannot find module 'debug'

### Operating System

ubuntu 20.01

### Programming Languages

JavaScript

### CCXT Version

4.1.73

### Description

node: 20.1.0


/home/gok123/.nvm/versions/node/v20.1.0/bin/node ./exchanges/Test.js
Uncaught Error Error: Cannot find module 'debug'
Require stack:
- /media/gok123/765844E85844A927/programowanie/javaScript/nodejs/node_modules/ccxt/dist/cjs/src/static_dependencies/proxies/http-proxy-agent/index.js
- /media/gok123/765844E85844A927/programowanie/javaScript/nodejs/node_modules/ccxt/dist/cjs/src/base/Exchange.js
- /media/gok123/765844E85844A927/programowanie/javaScript/nodejs/node_modules/ccxt/dist/cjs/ccxt.js
- /media/gok123/765844E85844A927/programowanie/javaScript/nodejs/node_modules/ccxt/dist/ccxt.cjs
- /media/gok123/765844E85844A927/programowanie/javaScript/Comparison/MakerComparision/exchanges/Test.js
    at Module._resolveFilename (internal/modules/cjs/loader:1082:15)
    at Module._load (internal/modules/cjs/loader:928:27)
    at Module.require (internal/modules/cjs/loader:1149:19)
    at require (internal/modules/helpers:121:18)
    at <anonymous> (/media/gok123/765844E85844A927/programowanie/javaScript/nodejs/node_modules/ccxt/dist/cjs/src/static_dependencies/proxies/http-proxy-agent/index.js:7:19)
    at Module._compile (internal/modules/cjs/loader:1267:14)
    at Module._extensions..js (internal/modules/cjs/loader:1321:10)
    at Module.load (internal/modules/cjs/loader:1125:32)
    at Module._load (internal/modules/cjs/loader:965:12)
    at Module.require (internal/modules/cjs/loader:1149:19)
    at require (internal/modules/helpers:121:18)
    at <anonymous> (/media/gok123/765844E85844A927/programowanie/javaScript/nodejs/node_modules/ccxt/dist/cjs/src/base/Exchange.js:662:87)
No debugger available, can not send 'variables'
Process exited with code 1

Old CCXT works


### Code

```
  
const ccxt = require("../../../nodejs/node_modules/ccxt/dist/ccxt.cjs");


```

--- a/ts/src/static_dependencies/proxies/http-proxy-agent/index.ts
+++ b/ts/src/static_dependencies/proxies/http-proxy-agent/index.ts
@@ -1,12 +1,12 @@
 import * as net from 'net';
 import * as tls from 'tls';
 import * as http from 'http';
-import createDebug from 'debug';
+// import createDebug from 'debug';
 import { once } from 'events';
 import type { OutgoingHttpHeaders } from 'http';
 import { Agent, AgentConnectOpts } from './../agent-base/index.js';
 
-const debug = createDebug('http-proxy-agent');
+// const debug = createDebug('http-proxy-agent');
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 type Protocol<T> = T extends `${infer Protocol}:${infer _}` ? Protocol : never;
@@ -58,7 +58,7 @@ export class HttpProxyAgent<Uri extends string> extends Agent {
 		super(opts);
 		this.proxy = typeof proxy === 'string' ? new URL(proxy) : proxy;
 		this.proxyHeaders = opts?.headers ?? {};
-		debug('Creating new HttpProxyAgent instance: %o', this.proxy.href);
+		// debug('Creating new HttpProxyAgent instance: %o', this.proxy.href);
 
 		// Trim off the brackets from IPv6 addresses
 		const host = (this.proxy.hostname || this.proxy.host).replace(
@@ -125,10 +125,10 @@ export class HttpProxyAgent<Uri extends string> extends Agent {
 		// Create a socket connection to the proxy server.
 		let socket: net.Socket;
 		if (this.secureProxy) {
-			debug('Creating `tls.Socket`: %o', this.connectOpts);
+			// debug('Creating `tls.Socket`: %o', this.connectOpts);
 			socket = tls.connect(this.connectOpts);
 		} else {
-			debug('Creating `net.Socket`: %o', this.connectOpts);
+			// debug('Creating `net.Socket`: %o', this.connectOpts);
 			socket = net.connect(this.connectOpts);
 		}
 
@@ -137,18 +137,18 @@ export class HttpProxyAgent<Uri extends string> extends Agent {
 		// to re-generate the string since we just changed the `req.path`.
 		let first: string;
 		let endOfHeaders: number;
-		debug('Regenerating stored HTTP header string for request');
+		// debug('Regenerating stored HTTP header string for request');
 		req._implicitHeader();
 		if (req.outputData && req.outputData.length > 0) {
 			// Node >= 12
-			debug(
-				'Patching connection write() output buffer with updated header'
-			);
+			// debug(
+			//	'Patching connection write() output buffer with updated header'
+			// );
 			first = req.outputData[0].data;
 			endOfHeaders = first.indexOf('\r\n\r\n') + 4;
 			req.outputData[0].data =
 				req._header + first.substring(endOfHeaders);
-			debug('Output buffer: %o', req.outputData[0].data);
+			// debug('Output buffer: %o', req.outputData[0].data);
 		}
 
 		// Wait for the socket's `connect` event, so that this `callback()`

--- a/ts/src/static_dependencies/proxies/https-proxy-agent/index.ts
+++ b/ts/src/static_dependencies/proxies/https-proxy-agent/index.ts
@@ -2,12 +2,12 @@ import * as net from 'net';
 import * as tls from 'tls';
 import * as http from 'http';
 import assert from 'assert';
-import createDebug from 'debug';
+// import createDebug from 'debug';
 import type { OutgoingHttpHeaders } from 'http';
 import { Agent, AgentConnectOpts } from './../agent-base/index.js';
 import { parseProxyResponse } from './parse-proxy-response.js';
 
-const debug = createDebug('https-proxy-agent');
+// const debug = createDebug('https-proxy-agent');
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 type Protocol<T> = T extends `${infer Protocol}:${infer _}` ? Protocol : never;
@@ -56,7 +56,7 @@ export class HttpsProxyAgent<Uri extends string> extends Agent {
 		this.options = { path: undefined };
 		this.proxy = typeof proxy === 'string' ? new URL(proxy) : proxy;
 		this.proxyHeaders = opts?.headers ?? {};
-		debug('Creating new HttpsProxyAgent instance: %o', this.proxy.href);
+		// debug('Creating new HttpsProxyAgent instance: %o', this.proxy.href);
 
 		// Trim off the brackets from IPv6 addresses
 		const host = (this.proxy.hostname || this.proxy.host).replace(
@@ -94,10 +94,10 @@ export class HttpsProxyAgent<Uri extends string> extends Agent {
 		// Create a socket connection to the proxy server.
 		let socket: net.Socket;
 		if (secureProxy) {
-			debug('Creating `tls.Socket`: %o', this.connectOpts);
+			// debug('Creating `tls.Socket`: %o', this.connectOpts);
 			socket = tls.connect(this.connectOpts);
 		} else {
-			debug('Creating `net.Socket`: %o', this.connectOpts);
+			// debug('Creating `net.Socket`: %o', this.connectOpts);
 			socket = net.connect(this.connectOpts);
 		}
 
@@ -144,7 +144,7 @@ export class HttpsProxyAgent<Uri extends string> extends Agent {
 			if (opts.secureEndpoint) {
 				// The proxy is connecting to a TLS server, so upgrade
 				// this socket connection to a TLS connection.
-				debug('Upgrading socket connection to TLS');
+				// debug('Upgrading socket connection to TLS');
 				const servername = opts.servername || opts.host;
 				return tls.connect({
 					...omit(opts, 'host', 'path', 'port'),
@@ -174,7 +174,7 @@ export class HttpsProxyAgent<Uri extends string> extends Agent {
 
 		// Need to wait for the "socket" event to re-play the "data" events.
 		req.once('socket', (s: net.Socket) => {
-			debug('Replaying proxy buffer for failed request');
+			// debug('Replaying proxy buffer for failed request');
 			assert(s.listenerCount('data') > 0);
 
 			// Replay the "buffered" Buffer onto the fake `socket`, since at

--- a/ts/src/static_dependencies/proxies/https-proxy-agent/parse-proxy-response.ts
+++ b/ts/src/static_dependencies/proxies/https-proxy-agent/parse-proxy-response.ts
@@ -1,8 +1,8 @@
-import createDebug from 'debug';
+// import createDebug from 'debug';
 import { IncomingHttpHeaders } from 'http';
 import { Readable } from 'stream';
 
-const debug = createDebug('https-proxy-agent:parse-proxy-response');
+// const debug = createDebug('https-proxy-agent:parse-proxy-response');
 
 export interface ConnectResponse {
 	statusCode: number;
@@ -35,16 +35,16 @@ export function parseProxyResponse(
 		}
 
 		function onclose(err?: Error) {
-			debug('onclose had error %o', err);
+			// debug('onclose had error %o', err);
 		}
 
 		function onend() {
-			debug('onend');
+			// debug('onend');
 		}
 
 		function onerror(err: Error) {
 			cleanup();
-			debug('onerror %o', err);
+			// debug('onerror %o', err);
 			reject(err);
 		}
 
@@ -57,7 +57,7 @@ export function parseProxyResponse(
 
 			if (endOfHeaders === -1) {
 				// keep buffering
-				debug('have not received end of HTTP headers yet...');
+				// debug('have not received end of HTTP headers yet...');
 				read();
 				return;
 			}
@@ -88,7 +88,7 @@ export function parseProxyResponse(
 					headers[key] = value;
 				}
 			}
-			debug('got proxy server response: %o', firstLine);
+			// debug('got proxy server response: %o', firstLine);
 			cleanup();
 			resolve({
 				connect: {

